app/cmd/controller.go

```diff
@@ -210,7 +210,9 @@ func startController(c *cli.Context) error {
 	control.GRPCAddress = util.GetGRPCAddress(listen)
 	control.GRPCServer = controllerrpc.GetControllerGRPCServer(volumeName, engineInstanceName, control)
 
-	control.StartGRPCServer()
+	if err = control.StartGRPCServer(); err != nil {
+		return err
+	}
```

Comment:
I think this change does not allow us to execute `control.Shutdown` in the event of a failure in `control.StartGRPCServer`. We return instead of entering `control.WaitForShutdown`, which would normally wait for the controller to intercept a signal and execute the registered `control.Shutdown` function. I think it's likely @shuo-wu intended to fall through to `control.WaitForShutdown`, even if `control.StartGRPCServer` failed (though it's weird we return, but do not log a couple of errors). Do you agree?

It looks like it would be ok to either:

- Maintain existing behavior by logging warnings in `control.StartGRPCServer` instead of allowing it to return an error, or
- Maintain existing behavior by logging a warning if `control.StartGRPCServer` returns an error, or
- Directly execute `control.Shutdown` if `control.StartGRPCServer` returns an error.